tests/integration/schema/migrations/query/with_p2p_schema_branch_test.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2023 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package query
+
+import (
+	"testing"
+
+	"github.com/lens-vm/lens/host-go/config/model"
+	"github.com/sourcenetwork/immutable"
+
+	"github.com/sourcenetwork/defradb/client"
+	testUtils "github.com/sourcenetwork/defradb/tests/integration"
+	"github.com/sourcenetwork/defradb/tests/lenses"
+)
+
+func TestSchemaMigrationQueryWithP2PReplicatedDocOnOtherSchemaBranch(t *testing.T) {
```

Comment:
thought: This test should propably be in the net replicator test no?

Reply:
It fits in both directories just as well IMO - I probably prefer it here because I see it more from a schema migration perspective, and you probably see it in the `net` directory as you see it more from a P2P perspective.

Reply:
In a sense, everything that is in the `net` directory also relate to some other test directory. In my mind, as soon as a test is interacting with the P2P system, it would be better placed in the `net` package.